common/src/main/java/io/druid/timeline/VersionedIntervalTimeline.java

```diff
@@ -265,6 +265,40 @@ public void add(final Interval interval, VersionType version, PartitionChunk<Obj
     }
   }
 
+  public boolean isOvershadowed(Interval interval, VersionType version)
+  {
```

Comment:
other methods in this class are protected by a readLock, 
I think we need to have similar protection here.